### PR TITLE
feat: Format code blocks at build time

### LINKED ISSUE #9057: Format code blocks during build for consistent styling
In the past, we had prettier in the repo which had the positive side effect that all (JS) code blocks in the docs had consistent formatting. Since we identified that prettier does not work well with current MDX we will lose this benefit.

However, It should be possible to format the code blocks during build, regardless of the input. With this I don't mean to actually modify commited code, but to sortof transpile the code blocks during the build, so that they are displayed formatted on the website.

We could apply different formatters for different languages and we may even use prettier for the JS code blocks.

### PR DESCRIPTION
**Pull request by a-hariti**
## Pre-merge checklist

*If you work at Sentry, you're able to merge your own PR without review, but please don't unless there's a good reason.*

- [ ] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## Description of changes

Format code blocks at build time in a transpilation fashion using Prettier

Ignores code blocks with `diff` metadata to avoid confusion due to leading `+` and `-` characters

Right now supports languages that are supported of the box by Prettier, there's room for more support for other languages through plugins

resolves #9057 

- [Sentry Docs contributor guide](https://docs.sentry.io/contributing/)

### CODE REVIEW

File: src/remark-format-code.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  };
+  const parser = parsers[lang];
+  if (!parser) {
+    return;
+  }
+
+  try {
+    const formattedCode = await format(node.value, {parser: parsers[lang]});
+    // get rid of the trailing newline
+    node.value = formattedCode.trimEnd();
```
Comment (lforst) [inline src/remark-format-code.js:42]: Instead of manually trimming the end, could we pass an option to `format?`

Comment (a-hariti) [reply src/remark-format-code.js]: Probably not, [options](https://prettier.io/docs/en/options)

**Comment by a-hariti** [comment]
PHP plugin seems to choke on the top level `public` keyword on [this line](https://github.com/getsentry/sentry-docs/blob/master/docs/platforms/php/guides/laravel/index.mdx#L25) although it's a [normal pattern](https://laravel.com/docs/10.x/errors#reporting-exceptions) in Laravel

I will wait on this for now until we have a clear idea of the preferred code formatting conventions among PHP developers

**Review by lforst — COMMENTED** [review]
If this works :shipit:

**Review by lforst — APPROVED** [review]

**Pull request merged**